### ISSUE TITLE
Error: pull access denied for <namedimage>, repository does not exist or may require 'docker login'

### ISSUE THREAD

**Issue by steveendow** [microsoft/navcontainerhelper#1158]
**UPDATE**: This appears to be the[ same error](https://github.com/microsoft/navcontainerhelper/issues/1055) as Issue #1055 

I have previously created named images and BC containers without issue on Docker EE v19.03.4 and 19.03.5.

I recently upgraded to Docker Engine Enterprise v19.03.11.

Following the Docker EE upgrade, I now get an error when trying to create a container, **based on a NEW named image**, using the imageName parameter.  It seems that the named image is created successfully, but when New-BcContainer tries to use the new named image to create a new container, that step fails with the error below.  

Since Issue #1055 had the error with Docker EE 19.03.5, it doesn't seem that the new Docker EE version is the cause of the problem.  Might I be doing something wrong?

Hmmm, I just realized that I have two versions of NavContainerHelper installed.  Could that be a factor?

```
PS C:\> get-installedmodule navcontainerhelper -allversions

Version    Name                                Repository           Description
-------    ----                                ----------           -----------
0.7.0.17   navcontainerhelper                  PSGallery            PowerShell module
0.7.0.18   navcontainerhelper                  PSGallery            PowerShell module
```

Here's the error:

```
DockerDo : Error response from daemon: pull access denied for mybcimage, repository does not exist or may require 'docker 
login': denied: requested access to the resource is denied
ExitCode: 1
Commandline: docker pull  mybcimage:sandbox-16.3.14085.14947-us
At C:\Program Files\WindowsPowerShell\Modules\navcontainerhelper\0.7.0.18\ContainerHandling\New-NavContainer.ps1:603 char:13
+             DockerDo -command pull -imageName $bestImageName | Out-Nu ...
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,DockerDo
```

If I re-run the exact same New-BcContainer script after this error occurs, it detects that the named image is present, and it successfully creates a container based on the named image.  No error.

```
$StartTime = $(get-date)

$containerName = 'test'
$password = 'P@ssw0rd'
$securePassword = ConvertTo-SecureString -String $password -AsPlainText -Force
$credential = New-Object pscredential 'admin', $securePassword
$auth = 'UserPassword'
$artifactUrl = Get-BcArtifactUrl -type 'Sandbox' -version '' -country 'us' -select 'Latest'
New-BcContainer `
    -accept_eula `
    -containerName $containerName `
    -credential $credential `
    -auth $auth `
    -artifactUrl $artifactUrl `
    -imageName 'mybcimage' `
    -dns '192.168.25.21' `
    -updateHosts

$elapsedTime = $(get-date) - $StartTime
$totalTime = "{0:HH:mm:ss}" -f ([datetime]$elapsedTime.Ticks)
write-host "Elapsed time:" $totalTime
```

```
PS C:\> C:\Users\sendow\Desktop\New BC Test Container with Elapsed time.ps1
NavContainerHelper is version 0.7.0.18
NavContainerHelper is running as administrator
Host is Microsoft Windows Server 2019 Standard - ltsc2019
Docker Client Version is 19.03.11
Docker Server Version is 19.03.11
Fetching all docker images
ArtifactUrl and ImageName specified
Downloading application artifact /sandbox/16.3.14085.14947/us
Downloading C:\Users\sendow\AppData\Local\Temp\e21c467e-7663-4f7b-a731-4f4e814201bf.zip
Unpacking application artifact to tmp folder using Expand-Archive
Image mybcimage:sandbox-16.3.14085.14947-us doesn't exist
Building image mybcimage:sandbox-16.3.14085.14947-us based on https://bcartifacts.azureedge.net/sandbox/16.3.14085.14947/us
Pulling latest image mcr.microsoft.com/dynamicsnav:10.0.17763.1339-generic
10.0.17763.1339-generic: Pulling from dynamicsnav
Digest: sha256:a1825dc0693031e86158c4805399d677878739dfe39572f0bea7330b38c34c19
Generic Tag: 0.1.0.7
Container OS Version: 10.0.17763.1339 (ltsc2019)
Host OS Version: 10.0.17763.1339 (ltsc2019)
Downloading platform artifact /sandbox/16.3.14085.14947/platform
Downloading C:\Users\sendow\AppData\Local\Temp\961eacf7-4126-42e4-84c0-c9b16962126c.zip
Unpacking platform artifact to tmp folder using Expand-Archive
Downloading Prerequisite Components
Downloading c:\bcartifacts.cache\sandbox\16.3.14085.14947\platform\Prerequisite Components\Open XML SDK 2.5 for Microsoft Office
\OpenXMLSDKv25.msi
Downloading c:\bcartifacts.cache\sandbox\16.3.14085.14947\platform\Prerequisite Components\IIS URL Rewrite Module\rewrite_2.0_rt
w_x64.msi
Downloading c:\bcartifacts.cache\sandbox\16.3.14085.14947\platform\Prerequisite Components\DotNetCore\DotNetCore.1.0.4_1.1.1-Win
dowsHosting.exe
Copying Platform Artifacts
Copy Database
Copy Licensefile
Copy ConfigurationPackages
Copy Extensions
Copy Applications.*
c:\bcartifacts.cache\tmp637311788498733832
Sending build context to Docker daemon  1.756GB

Step 1/6 : FROM mcr.microsoft.com/dynamicsnav:10.0.17763.1339-generic
 ---> 1926be7ad498
Step 2/6 : ENV DatabaseServer=localhost DatabaseInstance=SQLEXPRESS DatabaseName=CRONUS IsBcSandbox=Y artifactUrl=https://bcarti
facts.azureedge.net/sandbox/16.3.14085.14947/us
 ---> Running in b7bf050ad4b5
Removing intermediate container b7bf050ad4b5
 ---> 05d61268628f
Step 3/6 : COPY my /run/
 ---> 631c4cc68dee
Step 4/6 : COPY NAVDVD /NAVDVD/
 ---> 8f01b704a659
Step 5/6 : RUN \Run\start.ps1 -installOnly
 ---> Running in 56689f278d85
Installing Business Central
Installing Url Rewrite
Installing OpenXML
Installing DotNetCore
Starting Local SQL Server
WARNING: Waiting for service 'SQL Server (SQLEXPRESS) (MSSQL$SQLEXPRESS)' to 
start...
WARNING: Waiting for service 'SQL Server (SQLEXPRESS) (MSSQL$SQLEXPRESS)' to 
start...
Starting Internet Information Server
Copying Service Tier Files
Copying Web Client Files
Copying Client Files
Copying ModernDev Files
Copying PowerShell Scripts
Copying ConfigurationPackages
Copying Test Assemblies
Copying Extensions
Copying Applications
Copying Applications.*
Copying ReportBuilder
Determining Database Collation
Changing Database Server Collation to Latin1_General_100_CS_AS
Restoring CRONUS Demo Database
Modifying Business Central Service Tier Config File for Docker
Creating Business Central Service Tier
Installing SIP crypto provider: 'C:\Windows\System32\NavSip.dll'
Starting Business Central Service Tier
Importing CRONUS license file
Stopping Business Central Service Tier
Installation took 926 seconds
Installation complete
Removing intermediate container 56689f278d85
 ---> ee642c073614
Step 6/6 : LABEL legal="http://go.microsoft.com/fwlink/?LinkId=837447"       created="202007241618"       nav=""       cu=""    
   country="US"       version="16.3.14085.14947"       platform="16.0.14073.14904"
 ---> Running in 0783247de422
Removing intermediate container 0783247de422
 ---> 97d29fd39f71
Successfully built 97d29fd39f71
Successfully tagged mybcimage:sandbox-16.3.14085.14947-us
Building image took 1814 seconds
Pulling image mybcimage:sandbox-16.3.14085.14947-us
DockerDo : Error response from daemon: pull access denied for mybcimage, repository does not exist or may require 'docker 
login': denied: requested access to the resource is denied
ExitCode: 1
Commandline: docker pull  mybcimage:sandbox-16.3.14085.14947-us
At C:\Program Files\WindowsPowerShell\Modules\navcontainerhelper\0.7.0.18\ContainerHandling\New-NavContainer.ps1:603 char:13
+             DockerDo -command pull -imageName $bestImageName | Out-Nu ...
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,DockerDo
```

After the error occurred, I was able to create two new containers based on the named image, without any issues.  The containers seem to work fine.

![image](https://user-images.githubusercontent.com/11199908/88417551-f9ebf380-cd96-11ea-9449-095210c0aa9f.png)

**Comment by freddydk** [comment]
sorry, this regression was fixed in 0.7.0.19

https://www.powershellgallery.com/packages/navcontainerhelper/0.7.0.19
0.7.0.19
Issue #1150 alc fails if project path ends in a \
Issue #1153 regression - Test-BcContainer fails due to container name clashes
Adding parameter sqlMemoryLimit to New-BcContainer (can be set like 50%, 1.5G or 1500M)
Regression: autobuilding images will fail trying to pull images it just built

**Comment by steveendow** [comment]
Updating to 0.7.0.19 eliminated the error!   Thanks!
